python/dlr/counter/publisher.py

```diff
@@ -0,0 +1,57 @@
+import concurrent.futures
+import queue
+import logging
+
+from .utils import resturlutils
+from . import config
+
+
+# Singleton class
+class MsgPublisher(object):
+    _stop_processing = False
+    _instance = None
+    resp_cnt = 0
+
+    @staticmethod
+    def get_instance():
+        """return single instance of class"""
+        if MsgPublisher._instance is None:
+            MsgPublisher._instance = MsgPublisher()
+        return MsgPublisher._instance
+
+    def __init__(self):
+        try:
+            self.client = resturlutils.RestUrlUtils()
+            self.record_queue = queue.Queue(maxsize=config.CALL_HOME_PUBLISH_MESSAGE_MAX_QUEUE_SIZE)
+            # start loop
+            self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=config.CALL_HOME_MAX_WORKERS_THREADS)
+            self.executor.submit(self._process_queue)
+            logging.info("msg publisher thread pool execution started")
+        except Exception as e:
+            logging.exception("msg publisher thread pool not started", exc_info=True)
+
+    def send(self, data):
+        try:
+            self.record_queue.put(data, block=False)
+        except queue.Full as e:
+            logging.exception("ccm msg publisher queue full")
+        except Exception as e:
+            logging.exception("unable to record messages in queue", exc_info=True)
+
+    def _process_queue(self):
+        while not MsgPublisher._stop_processing:
+            if MsgPublisher.resp_cnt < config.CALL_HOME_REQ_STOP_MAX_COUNT:
+                resp_code = self.client.send(self.record_queue.get(block=True, timeout=2))
```

Comment:
Please add try catch for timeout exception

Reply:
Added appropriate exception handling.